lib/rubocop/cop/lint/empty_file.rb

```diff
@@ -21,14 +21,24 @@ module Lint
       #   # File consisting only of comments
       #
       class EmptyFile < Base
+        extend AutoCorrector
+
         MSG = 'Empty file detected.'
 
         def on_new_investigation
-          add_global_offense(MSG) if offending?
+          return unless offending?
+
+          add_global_offense(MSG) do
+            autocorrect if autocorrect_requested?
+          end
         end
 
         private
 
+        def autocorrect
+          FileUtils.rm_f(processed_source.file_path)
```

Comment:
It might be a very rate edge case, but I have a feeling this should not be executed when RuboCop processes stdin, e.g. via

```
echo -n '' | bundle exec rubocop --stdin <path> --only Lint/EmptyFile
```